### PR TITLE
PUBDEV-5613 Ensure AutoML respects max_runtime_secs

### PR DESCRIPTION
**Pull request by navdeep-G**
* This PR ensures AutoML respects the `max_runtime_secs` parameter by always checking the amount of time remaining in every poll of a job, which in this case is a model build.
* This PR also adds in a case for an empty leaderboard, which returns an empty leaderboard that has binomial metrics (we can change the headers to something else if needed)
* This PR also updates each test that uses a small `max_runtime_secs` parameter by always using `max_models` >=2 to avoid issues if the leaderboard is empty due to lack of run time.

### CODE REVIEW

File: h2o-automl/src/main/java/ai/h2o/automl/Leaderboard.java [hunk 1]
```diff
@@ -732,7 +731,11 @@ public TwoDimTable toTwoDimTable(String tableHeader, boolean leftJustifyModelIds
     //long[] timestamps = getTimestamps(models);
     String[] modelIDsFormatted = new String[models.length];
 
-    if(models[0]._output.isBinomialClassifier()) {
```
Comment (michalkurka) [inline h2o-automl/src/main/java/ai/h2o/automl/Leaderboard.java]: Can we tell if we are working on a binomial problem without having the first model?  Would be more consistent.

File: h2o-automl/src/main/java/ai/h2o/automl/Leaderboard.java [hunk 2]
```diff
@@ -653,7 +652,7 @@ public static final TwoDimTable makeTwoDimTable(String tableHeader, String sort_
       return new TwoDimTable(tableHeader,
               "no models in this leaderboard",
               rowHeaders,
-              Leaderboard.colHeaders(sort_metric, other_metrics),
```
Comment (michalkurka) [inline h2o-automl/src/main/java/ai/h2o/automl/Leaderboard.java]: Is this intended? How does it relate to this PR?

Comment (navdeep-G) [reply h2o-automl/src/main/java/ai/h2o/automl/Leaderboard.java]: Yes, it is intended just to put the point across no leaderboard is binomial.

File: h2o-automl/src/main/java/ai/h2o/automl/Leaderboard.java (hunk 1 above)
Comment (navdeep-G) [reply h2o-automl/src/main/java/ai/h2o/automl/Leaderboard.java]: Not sure. We could before when we just had a single sort metric for each model type. But now a user can pass in any.

File: h2o-automl/src/main/java/ai/h2o/automl/AutoML.java [hunk 3]
```diff
@@ -693,11 +692,6 @@ private boolean exceededSearchLimits(String whatWeAreSkipping) {
       return true;
     }
 
-    if (timeRemainingMs() <= 0.001) {
```
Comment (michalkurka) [inline h2o-automl/src/main/java/ai/h2o/automl/AutoML.java]: Do we need to remove this? Might be better not to start the modeling if we know for sure we will run out of time anyway

Comment (navdeep-G) [reply h2o-automl/src/main/java/ai/h2o/automl/AutoML.java]: I removed it bc I only want a single place of checking if time has run out. I can add it back, but the polling of the job should pick this up.

File: h2o-automl/src/main/java/ai/h2o/automl/Leaderboard.java (hunk 1 above)
Comment (michalkurka) [reply h2o-automl/src/main/java/ai/h2o/automl/Leaderboard.java]: It seems like each model makes the decision what family to use itself AutoML doesn't really know what problem is being solved based on the code. In that case this is fine.

**Pull request merged**